backend/core/src/auth/passport-strategies/jwt.strategy.ts

```diff
@@ -35,9 +36,15 @@ export class JwtStrategy extends PassportStrategy(Strategy) {
       throw new UnauthorizedException()
     }
     const userId = payload.sub
-    return this.userRepository.findOne({
+    const user = await this.userRepository.findOne({
       where: { id: userId },
       relations: ["leasingAgentInListings"],
     })
+
+    if (UserService.isPasswordOutdated(user)) {
+      throw new UnauthorizedException()
```

Comment:
We want to return a message to the user that lets them know that their password has expired. Can you update this to use `HttpException` and add an entry with translation string for `USER_ERRORS`. The frontend swallows this right now, but we can address that in the frontend issue for this (which will be #2017).

Reply:
Currently we have two ways to update the password:

1. PUT /user/:id (requires access token)
2. PUT /user/update-password (part of forgot password flow)

> since a user cannot login if their password is expired, update-password won't work, because it currently requires a token. The current update password endpoint is still valid, because we want users to be able to reset their password while logged in.

I was under an impression that we want to use forgot password flow when user account becomes locked because password is outdated. User requests forgot pasword email, receives reset URL, clicks the URL and sets up new password. 

I'm OK with implementing another endpoint only for resettting the password, if this is not sufficient. Please let me know how to proceed with this.

Reply:
Ah, yes, forgot password flow will work fine.